### DISCUSSION TITLE
Why Doesn't Supabase Auth Respect emailRedirectTo values?

### DISCUSSION THREAD

**Discussion by ganeshrvel** [supabase/supabase#30796] [Questions] [answered]
I'm working on a Supabase-based project and trying to implement custom redirect URLs for authentication in a development environment. Specifically, I want users to be redirected to `http://localhost:3000/auth/confirm` after they sign in with an OTP. Here's the setup:  

### Email Template  
```html
<a href="{{ .RedirectTo }}?email={{ .Email }}&code={{ .Token }}">
```

### Supabase Client Setup  
```typescript
import { createServerClient } from '@supabase/ssr';
export async function supabaseInstance(): Promise<SupabaseClient> {
  const cookieStore = await cookies();

  return createServerClient(
    NEXT_PUBLIC_SUPABASE_URL,
    NEXT_PUBLIC_SUPABASE_ANON_KEY,
    {
      cookies: {
        getAll() {
          return cookieStore.getAll();
        },
        setAll(cookiesToSet) {
          cookiesToSet.forEach(({ name, value, options }) =>
            cookieStore.set(name, value, options),
          );
        },
      },
    },
  );
}
```

### Sign-In Implementation  
```typescript
const supabase = await supabaseAuthOnlyServerInstance();
const signInOptions = {
  email: email,
  shouldCreateUser: true,
  emailRedirectTo: 'http://localhost:3000/auth/confirm',
};

const { error } = await supabase.auth.signInWithOtp(signInOptions);
```

### Redirect URL Configurations in Supabase Dashboard  
I've tried adding the following patterns under Authentication settings:  
```
http://localhost:3000/**
http://localhost:3000/*
http://**localhost:3000/*/*
http://**localhost:3000/*
http://**localhost:3000/**
http://localhost:3000
http://localhost**
http://localhost*
https://app.example.com/**
```

Despite this, the redirection always defaults to the `SITE URL`, configured as `https://app.example.com/confirm/auth`, on Supabase dashboard. The `emailRedirectTo` parameter seems to be ignored entirely.  
Even if I set the `emailRedirectTo` as `https://app.example.com/signup`, the `emailRedirectTo` value isn't being honored

#### Are there any restrictions or special steps for making `emailRedirectTo` URLs work for both production as well as localhost?

Any help or guidance would be greatly appreciated!

**Comment by GaryAustin1** [comment, marked as answer]
I think options are under options:{}
![image](https://github.com/user-attachments/assets/b5f1e0ac-e78e-4f57-b7c9-17913394bf9e)

**Reply by ganeshrvel** [reply]
Oh wow. I missed that. Thanks!
